### ISSUE TITLE
MDNS_NAME_BUF_LEN need to be in include/mdns.h (IDFGH-14326)

### ISSUE THREAD

**Issue by jonsmirl** [espressif/esp-protocols#724]
### Answers checklist.

- [X] I have read the documentation for esp-protocols [components](https://github.com/espressif/esp-protocols#readme) and the issue is not addressed there.
- [X] I have updated my esp-protocols branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### General issue report

![image](https://github.com/user-attachments/assets/82464fe1-b88a-4c62-ab7d-f97ef3538d90)
![image](https://github.com/user-attachments/assets/5db9b518-28c2-4b69-90fe-2ab2c301142a)

**Comment by david-cermak** [comment]
This is correct, need to provide the maximum size in public headers if we require users to allocate buffers for this API.
Thank for the report, will fix.
